pos_product_sync/static/src/js/pos_product_sync.js

```diff
@@ -75,6 +79,15 @@ odoo.define('pos_product_sync.pos', function (require) {
 
             // Orderlines
             this.update_orderlines(product_ids);
+        },
+
+        update_product_related_templates_images: function(product_ids) {
```

Comment:
Strange, you update here everything except orderlines. Better leave one function as ``update_templates_related_to_product_fields`` and update images with ``update_image`` before  ``update_templates_related_to_product_fields``  is called on the line 25

Reply:
done